pyteal/ast/frame.py

```diff
@@ -0,0 +1,188 @@
+from typing import TYPE_CHECKING
+
+from pyteal.ast.expr import Expr
+from pyteal.types import TealType, require_type
+from pyteal.errors import TealInputError, verifyProgramVersion
+from pyteal.ir import TealBlock, TealSimpleBlock, TealOp, Op
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class Proto(Expr):
+    def __init__(self, arg_num: int, ret_num: int):
```

Comment:
```suggestion
    def __init__(self, num_args: int, num_returns: int):
```
slight rename suggested